wagtail_wordpress_import/test/tests/test_block_builder.py

```diff
@@ -1,11 +1,289 @@
+        ).find("img")
+        self.assertEqual(get_image_alt(input), "image alt")
+
+    def test_get_image_file_name(self):
+        self.assertEqual(get_image_file_name("fakeimage.jpg"), "fakeimage.jpg")
+        self.assertEqual(get_image_file_name("folder/fakeimage.jpg"), "fakeimage.jpg")
+        self.assertEqual(
+            get_image_file_name(
+                "http://www.example.com/folder1/folder2//fakeimage.jpg"
+            ),
+            "fakeimage.jpg",
+        )
+
+    def test_get_abolute_src(self):
+        self.assertEqual(
+            get_abolute_src("fakeimage.jpg", "http://www.example.com"),
+            "http://www.example.com/fakeimage.jpg",
+        )
+        self.assertEqual(
+            get_abolute_src("folder/fakeimage.jpg", "http://www.example.com"),
+            "http://www.example.com/folder/fakeimage.jpg",
+        )
+        self.assertEqual(
+            get_abolute_src("folder/fakeimage.jpg"),
+            "folder/fakeimage.jpg",
+        )  # the test settings has no BASE_URL setting so try having no domain prefix
+
+    def test_get_abolute_src_slashes_at_start(self):
+        self.assertEqual(
+            get_abolute_src("//folder/fakeimage.jpg", "http://www.example.com"),
+            "http://www.example.com/folder/fakeimage.jpg",
+        )
+
+    def test_get_alignment_class(self):
+        input = get_soup(
+            '<img src="fakeimage.jpg" alt="image alt" class="align-left" />',
+            "html.parser",
+        ).find("img")
+        self.assertEqual(get_alignment_class(input), "left")
+        input = get_soup(
+            '<img src="fakeimage.jpg" alt="image alt" class="align-right" />',
+            "html.parser",
+        ).find("img")
+        self.assertEqual(get_alignment_class(input), "right")
+        input = get_soup(
+            '<img src="fakeimage.jpg" alt="image alt" />',
+            "html.parser",
+        ).find("img")
+        self.assertEqual(get_alignment_class(input), "fullwidth")
+
+    # work in progress
+    def test_with_real_image(self):
+        # but we need to test with mocked images if we can.
+        raw_html_file = """
+        <p>Lorem <img src="https://dummyimage.com/600x400/000/fff" alt=""></p>
+        """
+        self.builder = BlockBuilder(raw_html_file, None, None)
+        self.builder.promote_child_tags()
+        self.blocks = self.builder.build()
+        self.assertTrue("<embed" in self.blocks[0]["value"])
+
+    # work in progress
+    def test_with_real_pdf(self):
+        # but we need to test with mocked files if we can.
+        raw_html_file = """
+        <p>
+        <a href="https://file-examples-com.github.io/uploads/2017/10/file-sample_150kB.pdf">A pdf</a>
+        </p>
+        """
+        self.builder = BlockBuilder(raw_html_file, None, None)
+        self.builder.promote_child_tags()
+        self.blocks = self.builder.build()
+        self.assertTrue('linktype="document"' in self.blocks[0]["value"])
```

Comment:
It's because we should really be testing without needed to use external files. I opened a ticket to look at this at some point rather than hold the PR's up. https://projects.torchbox.com/projects/wordpress-to-wagtail-importer-package/tickets/76